src/board.cpp

```diff
@@ -1,35 +1,44 @@
 #include "board.h"
 
 #define translateRow(row) std::max(0, row + BOARD::ROW_OFFSET)
-#define translateCol(col) std::max(0, std::min(col + BOARD::COLUMN_OFFSET, BOARD::TRUE_ROW_SIZE - 1)) // returns 0 or 11 if row + offset is not between 1 and 10
+#define translateCol(col) std::max(0, std::min(col + BOARD::COLUMN_OFFSET, BOARD::TRUE_ROW_SIZE - 1)) // returns 0 or 11 if !(0 < row + offset < 11)
 
 Board::Board() {
-	this->board = std::vector<int>(BOARD::TRUE_BOARD_SIZE, 0);
+	this->board = std::vector<BoardCell>(BOARD::TRUE_BOARD_SIZE);
 	this->blocksPerRow = std::vector<int>(BOARD::COLUMN_SIZE, 0);
 	this->currentHeight = 1;
+	int row = BOARD::LEFT_BOUNDARY, col = BOARD::BOTTOM_BOUNDARY;
+
+	for (int row = 0; row < BOARD::COLUMN_SIZE; ++row) {
+		for (int col = 0; col < BOARD::ROW_SIZE; ++col) {
+			Block block(row, col);
```

Comment:
I'm wary of your use of block here; it seems that cell stores a reference to block, which is a stack variable so it is out of scope. Consider changing this to dynamic allocation or not store a reference in BoardCell.